iceoryx2/src/service/builder/mod.rs

```diff
@@ -296,7 +296,7 @@ impl<ServiceType: service::Service> BuilderWithServiceType<ServiceType> {
                     .config(&static_config_storage_config::<ServiceType>(
                         self.global_config.as_ref(),
                     ))
-                    .has_ownership(false)
+                    .has_ownership(true)
```

Comment:
Ownership is released in the `builder/event.rs|publish_subscribe.rs` when the initialization is finished. If it fails the acquired ownership ensures that everything is cleaned up.